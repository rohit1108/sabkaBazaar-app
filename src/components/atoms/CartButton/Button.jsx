import React from "react";
import "./Button.scss";

const Button = ({ className, method, children, disabled, type , dataTestId}) => {
  return (
    <button
      className={className}
      onClick={method}
      type={type}
      disabled={disabled}
      data-testid={dataTestId}
    >
      {children}
    </button>
  );
};
export default Button;
