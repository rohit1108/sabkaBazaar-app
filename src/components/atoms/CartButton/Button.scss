.button {
  display: block;
  border-radius: 3px;
  text-align: center;
  font-size: 0.5rem;
  padding: 0.5rem;
  border: 1px solid #eee;
  background-color: #d42559;
  color: #fff;
  cursor: pointer;
  position: relative;
}
