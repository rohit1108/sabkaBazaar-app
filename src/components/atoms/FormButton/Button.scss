@mixin button-bg($bg) {
    background: $bg;
    border: none;

    &:hover {
        background: darken($bg, 8%);
        transition: all 0.3s ease;
    }

    &:active {
        background: darken($bg, 25%);
    }
}

.btn {
    color: white;
    padding: 5px 10px;
    border-radius: 1px;
}

.btn-red {
    @include button-bg(#e43774);
}