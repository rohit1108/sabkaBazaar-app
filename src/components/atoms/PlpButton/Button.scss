@import "../../../styles/variables";

// Default Class for Buttons
.button-wrapper {
  display: block;
  border-radius: 3px;
  text-align: center;
  text-decoration: none;
  font-size: $text-font-size;
  padding: 0.5rem;
  border: 1px solid #eee;
  background-color: $button-color;
  color: $button-text-color;
  cursor: pointer;
  position: relative;

  &:active {
    top: 0.6px;
    left: 0.6px;
  }

  &:disabled {
    background-color: $secondary-color;
    cursor: not-allowed;
  }

  @media screen and (min-width: $mobile-breakpoint) {
    font-size: $text-font-size;
  }
}
