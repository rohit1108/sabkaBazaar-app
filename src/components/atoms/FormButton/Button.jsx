import React from 'react'
import './Button.scss'

function Button({ children, onClick }) {

  return (
    <div>
      <button className="btn btn-red" onClick={onClick}>{children}</button>
    </div>
  )
}

export default Button