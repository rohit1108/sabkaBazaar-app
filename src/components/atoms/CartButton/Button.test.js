import React from 'react';
import { fireEvent, render, screen} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Button from './Button';

describe("Button component", () => {
    it('renders the component properly', () => {
        render(<Button/>);
    });

    it('should receive props correctly', () => {
        const fireSomeEvent = jest.fn();
        render(<Button children={"Button children"} 
            buttonType = ""
            method = {fireSomeEvent}
            id={"5b6c6aeb01a7c38429530884"} dataTestId="button"/>);
        expect(screen.getByTestId("button")).toHaveTextContent("Button children");
        const buttonElement = screen.getByRole("button", { name: "Button children" });
        fireEvent.click(buttonElement);
        expect(fireSomeEvent).toBeCalledTimes(1);

    });

    it('should have the right Button component snapshot', () => {
        const tree = renderer.create(<Button/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

});