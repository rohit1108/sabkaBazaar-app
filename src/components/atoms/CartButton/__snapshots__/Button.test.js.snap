// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should have right Button comp snapshot 1`] = `<button />`;

exports[`Button component should have the right Button component snapshot 1`] = `<button />`;

exports[`Sorting Component should have right Button comp snapshot 1`] = `<button />`;
